packages/tx/src/eip1559Transaction.ts

```diff
@@ -42,7 +44,10 @@ const TRANSACTION_TYPE_BYTES = hexToBytes(
  * - TransactionType: 2
  * - EIP: [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)
  */
-export class FeeMarketEIP1559Transaction extends BaseTransaction<TransactionType.FeeMarketEIP1559> {
+export class FeeMarketEIP1559Transaction
+  extends BaseTransaction<TransactionType.FeeMarketEIP1559>
+  implements EIP1559CompatibleTxInterface<TransactionType.FeeMarketEIP1559>
```

Comment:
I think this is not structurally correct to add this `implements` note respectively is both redundant and not complete at the same time. So I think we should remove.

If we would want to do this correctly/completely we would need to implement from several compatibility interfaces (here e.g. also from the EIP2930 interface), but multiple inheritance unfortunately doesn't exist in JavaScript (I thought a bit along these lines too some 1-2 years ago when thinking about how we could refactor this stuff).

Reply:
Yeah, I actually went back and forth on this. I agree about the point concerning multiple inheritance, but since all of the interfaces are nested within each other (eip1559 inherits from eip2930, and eip4844 inherits from eip1559), I figured this wasn't a problem. However I agree that this would not be "correct", especially if we want to treat capabilities as independent from one another.

There are no type errors even if we remove this `ìmplements` (typescript automatically figures out whether or not the class fits the necessary interface when we use the Capabilities, regardless of if we explicitly mention that it `implements` said interface. I'll go ahead and remove this.